### DISCUSSION TITLE
Running code in Sandbox

### DISCUSSION THREAD

**Discussion by dcieslak19973** [open-webui/open-webui#6272] [Q&A]
A fairly common prompt to demonstrate an LLM's ability to write code is something along the lines of "Using only Python's requests module, write a program that gathers the last year's stock prices for Nvidia and show them on a graph".

Assuming the LLM generates reasonable code, running it in the Sandbox fails, generally with something like:

```
PythonError: Traceback (most recent call last):
File "/lib/python3.12/site-packages/urllib3/contrib/emscripten/fetch.py", line 385, in send_request
js_xhr.send(to_js(request.body))
pyodide.ffi.JsException: NetworkError: Failed to execute 'send' on 'XMLHttpRequest': Failed to load 'https://query1.finance.yahoo.com/v7/finance/download/NVDA?period1=1697751469&period2=1729287469&interval=1d&events=history&includeAdjustedClose=true'.
```

Is there a way to allow the Sandbox to make network calls?

**Comment by mafzaal** [comment]
I think it's issue with yahoo finance. I have downloaded data using other services such as alphavantage.co
